src/Simplify.cpp

```diff
@@ -4659,8 +4682,10 @@ void check_boolean() {
 void check_math() {
     Var x = Var("x");
 
+    check(sqrt(4.0f), 2.0f);
     check(log(0.5f + 0.5f), 0.0f);
     check(exp(log(2.0f)), 2.0f);
+    check(pow(exp(1.0f), log(10.0f)), 10.0f);
```

Comment:
The rules look good, but this check looks sensitive to fp precision issues. Is it reasonable to expect that std::pow(std::exp(1.0f), std::log(10.0f)) == 10.0f exactly? Each computation in the folding is done in doubles, but it still seems like there's a bunch of intermediate rounding that could perturb the result.

Maybe: check(round(1000.0f*pow(exp(1.0f), log(10.0f))), 10000.0f); ? That should catch any bugs in the corresponding simplifier rules.